lib/active_merchant/billing/gateways/tns.rb

```diff
@@ -150,7 +157,12 @@ def commit(action, post)
           'Content-Type' => 'application/json',
         }
         post[:apiOperation] = action.upcase
-        raw = parse(ssl_request(:put, url, build_request(post), headers))
+        begin
+          raw = parse(ssl_request(:put, url, build_request(post), headers))
+        rescue ResponseError => e
+          raise unless(e.response.code.to_s =~ /400/)
```

Comment:
In looking at the `test_invalid_login` remote test... if that gets modified to no longer raise the exception, then I think this line won't be needed.